sdk/cosmosdb/cosmos/src/queryExecutionContext/pipelinedQueryExecutionContext.ts

```diff
@@ -45,11 +47,48 @@ export class PipelinedQueryExecutionContext implements ExecutionContext {
     if (this.pageSize === undefined) {
       this.pageSize = PipelinedQueryExecutionContext.DEFAULT_PAGE_SIZE;
     }
+
+    // Initialize continuation token manager early so it's available for OffsetLimitEndpointComponent
+    const sortOrders = partitionedQueryExecutionInfo.queryInfo.orderBy;
+    const isOrderByQuery = Array.isArray(sortOrders) && sortOrders.length > 0;
+    if (this.options.enableQueryControl) {
+      this.continuationTokenManager = new ContinuationTokenManager(
+        this.collectionLink,
+        this.options.continuationToken,
+        isOrderByQuery,
+      );
+    }
+
     // Pick between Nonstreaming and streaming endpoints
     this.nonStreamingOrderBy = partitionedQueryExecutionInfo.queryInfo.hasNonStreamingOrderBy;
 
+    // Check if this is a GROUP BY query
+    const isGroupByQuery =
+      Object.keys(partitionedQueryExecutionInfo.queryInfo.groupByAliasToAggregateType).length > 0 ||
+      partitionedQueryExecutionInfo.queryInfo.aggregates.length > 0 ||
+      partitionedQueryExecutionInfo.queryInfo.groupByExpressions.length > 0;
+
+    // Check if this is an unordered DISTINCT query
+    const isUnorderedDistinctQuery =
+      partitionedQueryExecutionInfo.queryInfo.distinctType === "Unordered";
+
+    // Configure continuation token manager for unsupported query types
+    if (
+      this.continuationTokenManager &&
+      (isUnorderedDistinctQuery || isGroupByQuery || this.nonStreamingOrderBy)
+    ) {
+      this.continuationTokenManager.setUnsupportedQueryType(true);
```

Comment:
Why do we have this method? 
can we avoid continuation token manager for unsupported queries?